docs/specifications/tests/Consistency-TP/consistency05.md

```diff
@@ -1,49 +1,205 @@
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+## Ordered description of steps to be taken to execute the test case
+1. Obtain the set of name server names from the NS records in the 
+   delegation of *Child Zone* using [Method2] and any glue IP addresses
+   from the same delegation using [Method4].
+
+   1. Extract the [in-bailiwick] name server names and create the set
+      "Delegation Strict Glue", where each name server name 
+      is matched with its IP address or addresses, if available. The 
+      set may be empty.
+
+   2. Extract the [out-of-bailiwick] name server names and create the 
+      set "Delegation Extended Glue", where each name server name 
+      is matched with its IP address or addresses, if available. The 
+      set may be empty.
+
+2. Obtain the set of name server names for the *Child Zone* using
+   [Method2] and [Method3] and extract the [in-bailiwick] name 
+   server names, "IB NS Name Set".
+
+3. Create an empty set of name server name with associated IP address
+   or addresses, "Address Records From Child".
+
+4. Obtain the set of name server IP addresses for *Child Zone* 
+   using [Method4] and [Method5], "NS IP", if *IB NS Name Set* is 
+   non-empty.
+
+5. If *IB NS Name Set* is non-empty, then for each name server name in
+   that set do:
+
+   1. Create one A query and one AAAA query with the RD flag unset
+      and name server name as owner name.
+
+   2. For each name server in *NS IP* and for each record 
+      types (A, AAAA):
+      1. Send the address query to the name server.
+      2. If there is no DNS response from the server, then
+         output *[NO_RESPONSE]*.
+      3. Or, if the response is a delegation (referral) to a 
+         sub-zone of *Child Zone*, then:
+         1. Copy the adress query (A, AAAA) that gave the referral
+            response.
+         2. Set the RD flag in the copied query (from unset to set).
+         3. Do a DNS lookup of the the query using a resolving name server 
+            on the public DNS.
+            * The lookup must take into account changes that
+              undelegated data has created, if any.
+         4. If the lookup returns the relevant address record or records
+            with the same owner name as in the query, then extract those 
+            and add to *Address Records From Child* with name and IP 
+            address or addresses.
+      4. Or, if the response has the AA flag unset, then
+         output *[CHILD_NS_FAILED]*. 
+      5. Or, if the RCODE of the response is neither NOERROR nor 
+         NXDOMAIN, then output *[CHILD_NS_FAILED]*.
+      6. Or, if the RCODE is NOERROR (with the AA flag set), then
+         extract any address records (A, AAAA) from the answer
+         section whose owner name matches the owner name 
+         of the query and add that or those to 
+         *Address Records From Child* with name and IP.
+      7. Or, if the RCODE is NXDOMAIN, there is nothing to do.
```

Comment:
Do these enumerated cases cover all possible inputs? If so, the condition here can be removed. If not, a default case should be provided.